src/main/java/net/sf/jabref/logic/util/io/FileFinder.java

```diff
@@ -1,43 +1,42 @@
 package net.sf.jabref.logic.util.io;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
 import java.util.Collection;
 import java.util.HashSet;
+import java.util.Objects;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-public class FileFinder {
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 
-    public static Set<File> findFiles(Collection<String> extensions, Collection<File> directories) {
-        Set<File> result = new HashSet<>();
+public class FileFinder {
 
-        for (File directory : directories) {
-            result.addAll(FileFinder.findFiles(extensions, directory));
-        }
+    private static final Log LOGGER = LogFactory.getLog(FileFinder.class);
+    private final static Set<File> result = new HashSet<>();
 
-        return result;
-    }
 
-    private static Collection<? extends File> findFiles(Collection<String> extensions, File directory) {
-        Set<File> result = new HashSet<>();
+    public static Set<File> findFiles(Collection<String> extensions, Collection<File> directories) {
 
-        File[] children = directory.listFiles();
-        if (children == null) {
-            return result; // No permission?
```

Comment:
Are there cases where the result is explicitly checked for an empty `Set` and some warning related to permissions is shown?